runtime/include/qio/qio.h

```diff
@@ -1040,6 +1041,7 @@ qioerr qio_channel_read_amt(const int threadsafe, qio_channel_t* restrict ch, vo
   if( qio_space_in_ptr_diff(len, ch->cached_end, ch->cached_cur) ) {
     qio_memcpy( ptr, ch->cached_cur, len );
     ch->cached_cur = qio_ptr_add(ch->cached_cur, len);
+    ch->mark_stack[ch->mark_cur] += len;
```

Comment:
I don't think this change is needed. The concept is that, when `ch->cached_cur` and `ch->cached_end` are in use, the current position is represented by `ch->cached_cur` itself. See e.g. `qio_channel_offset_unlocked` which shows how the offest is computed in this manner.

But, presumably there was some reason you added it. Could you address that concern in another way?

Reply:
Without this line, the next read was not picking up where the previous had left off (it seemed like `cached_cur` was not being used to represent the channel's position in the file). 

I'll try to figure out why and hopefully solve this problem in a more elegant way.

Reply:
Good point, that has been fixed now.